File: C++/judgePalindrome.cpp
#include<iostream>
#include<string>

class Solution{
    public:
    bool judge(std::string str) {
        int left = 0;
        int right = str.size() - 1;
        
        while (left < right) {
            if (str[left] != str[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};

int main(){
    std::string str1="absba";
    std::string str2="ranko";
    std::string str3="yamatomaya";
    std::string str4="a";
    Solution solution;
    std::cout<<solution.judge(str1)<<"\n"<<solution.judge(str2)<<"\n"<<solution.judge(str3)<<"\n"<<solution.judge(str4)<<"\n";
    return 0;
}

File: python/judgePalindrome.py
class Solution:
    def judge(self , str: str) -> bool:
        left=0
        right=len(str)-1
        while left<right:
            if str[left] !=str[right]:
                return False
            left +=1
            right -=1
        return True
    
s1="yamatomaya"
s=Solution()
print(s.judge(s1))